getspice.py

```diff
@@ -22,135 +20,222 @@
+from zipfile import ZipFile
 
 import six.moves.urllib as urllib
-import io
-import zipfile
-import subprocess
-
-__author__ = 'AndrewAnnex'
-
-def getSpice():
-
-    def give_points(dct, info):
-        for key in dct:
-            if info in key:
-                dct[key] += 1
-
-    def get_winner(dct):
-        candidates = list(dct.keys())
-        values = list(dct.values())
-        winner = candidates[0]
-        winner_value = 0
-        for cand, value in zip(candidates, values):
-            if value > winner_value:
-                winner_value = value
-                winner = cand
-        return winner
-
-    root_url = 'https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N0066/C/'
-    platform_urls = [
-        'MacIntel_OSX_AppleC_32bit/',
-        'MacIntel_OSX_AppleC_64bit/',
-        'MacPPC_OSX_AppleC_32bit/',
-        'PC_Cygwin_GCC_32bit/',
-        'PC_Linux_GCC_32bit/',
-        'PC_Linux_GCC_64bit/',
-        'PC_Windows_VisualC_32bit/',
-        'PC_Windows_VisualC_64bit/']
-    ### DETERMINE BEST DOWNLOAD OPTION ###
-    print('Gathering information...')
-    points = {url: 0 for url in platform_urls}
-
-    system = platform.system()
-    if system == 'Darwin':
-        system = 'Mac'
-    print('SYSTEM:   ', system)
-    give_points(points, system)
-
-    processor = platform.processor()
-    print('PROCESSOR:', processor)
 
-    machine = '64bit' if sys.maxsize > 2 ** 32 else '32bit'
-    print('MACHINE:  ', machine)
-    give_points(points, machine)
-
-    result = get_winner(points) + 'packages/cspice.tar.Z'
-
-    if "Windows" in result:
-        result = result.replace('.tar.Z', '.zip')
-
-    print('Best option:', result.split('/')[0])
-    print('URL: ', root_url + result)
-
-    ### DOWNLOAD AND UNPACK BEST PACKAGE ###
-    root_dir = os.path.realpath(os.path.dirname(__file__))
-
-    print('\nDownloading...')
-
-    attemptSpiceDownloadXTimes(10, root_url, result, root_dir)
+__author__ = 'JorgeDiazdelRio'
+
+
+# =============================================================================
+# Handling of CSPICE downloads from HTTPS
+# ---------------------------------------
+#    Some Python distributions may be linked to an old version of OpenSSL
+#    which will not let you connect to NAIF server due to recent SSL cert
+#    upgrades on the JPL servers.  Moreover, versions older than
+#    OpenSSL 1.0.1g are known to contain the 'the Heartbleed Bug'.  Therefore
+#    this module provides two different implementations for the function
+#    __download_ssl (which implements the HTTPS GET call to the NAIF server
+#    to download the required CSPICE distribution package).
+#
+def _download_ssl(url):
+    """
+    Returns the CSPICE package located at the NAIF server using the input
+    ``url``.  This fuction is implemented using urllib3 and pyOpenSSL in
+    order to resolve the known issue with OpenSSL MacOS Python
+    distributions before Python3.6.x
+
+    Parameters
+    ----------
+    url: str
```

Comment:
could you reformat the style of the docstring to match the RST/sphinx format I use for the rest of spiceypy?